src/pcap/bosh_resolver.go

```diff
@@ -0,0 +1,208 @@
+	}
+
+	return nil
+}
+
+func (br *BoshResolver) setup() error {
+	log.Infof("Setting Up BoshResolver for %s", br.environment.Alias)
+
+	if br.environment.CaCert == "" {
+		br.client = http.DefaultClient
+	} else {
+		data, err := os.ReadFile(br.environment.CaCert)
+		if err != nil {
+			return fmt.Errorf("could not load BOSH Director CA from %s (%s)", br.environment.CaCert, err)
+		}
+
+		boshCA := x509.NewCertPool()
+		ok := boshCA.AppendCertsFromPEM(data)
+
+		if !ok {
+			return fmt.Errorf("could not add BOSH Director CA from %s, adding to the cert pool failed.", br.environment.CaCert)
+		}
+
+		br.client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: boshCA,
+				},
+			},
+		}
+	}
+
+	log.Info("Discovering BOSH Director endpoint...")
+	response, err := br.client.Get(br.environment.RawDirectorURL + "/info")
+
+	if err != nil {
+		return fmt.Errorf("could not fetch BOSH Director API from %s (%s)", br.environment.RawDirectorURL, err)
+	}
+
+	var apiResponse *bosh.Info
+	data, err := io.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("could not read BOSH Director API response: %s", err)
+	}
+	err = json.Unmarshal(data, &apiResponse)
+	if err != nil {
+		return fmt.Errorf("could not parse BOSH Director API response: %s", err)
+	}
+
+	br.uaaURLs = apiResponse.UserAuthentication.Options.Urls
+
+	log.Infof("Connected to BOSH Director '%s' (%s), version %s on %s. UAA URLs: %v", apiResponse.Name, apiResponse.Uuid, apiResponse.Version, apiResponse.Cpi, br.uaaURLs)
+
+	return nil
+}
+
+func (br *BoshResolver) authenticate(authToken string) error {
+
+	allowed, err := VerifyJwt(authToken, "bosh.admin", br.uaaURLs)
+	if err != nil {
+		return fmt.Errorf("could not verify token %s (%s)", authToken, err)
+	}
+
+	if !allowed {
+		return fmt.Errorf("token %s does not have the permissions or is not supported", authToken)
+	}
+
+	return nil
+}
+
+func (br *BoshResolver) getInstances(deployment string, authToken string) ([]bosh.Instance, int, error) {
+	log.Debugf("Checking at %s if deployment %s can be seen by token %s", br.environment.RawDirectorURL, deployment, authToken)
+	instancesURL, err := url.Parse(fmt.Sprintf("%s/deployments/%s/instances", br.environment.RawDirectorURL, deployment))
+	if err != nil {
+		return nil, 0, err
+	}
+
+	req := &http.Request{
+		Method: http.MethodGet,
+		URL:    instancesURL,
+		Header: map[string][]string{
+			"Authorization": {"Bearer " + authToken},
+		},
+	}
+
+	res, err := br.client.Do(req)
+	if err != nil {
+		return nil, 0, fmt.Errorf("request to Bosh-director failed: %v", zap.Error(err))
+	}
+
+	defer res.Body.Close()
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return nil, res.StatusCode, fmt.Errorf("expected status code %d but got status code %d: %s", http.StatusOK, res.StatusCode, string(data))
+	}
+
+	var response []bosh.Instance
```

Comment:
not specific to this line, but having a package `bosh` for some of the types is a bit weird. Either put the whole resolver into a package `bash` or all code into the `bash.go` in package `pcap`.